src/app/shared/palette/address/write-address-field.component.ts

```diff
@@ -72,7 +72,10 @@ export class WriteAddressFieldComponent extends AbstractFieldWriteComponent impl
   }
 
   isAddressSet() {
-    const address = this.caseField.value;
+    if (!this.writeComplexFieldComponent || !this.writeComplexFieldComponent.complexGroup) {
+      return false;
+    }
+    const address = this.writeComplexFieldComponent.complexGroup.value;
```

Comment:
Do we need to have test coverage on this scenario?

Reply:
Adrian's right, it's failing the Sonar quality gate on coverage of new code: https://build.platform.hmcts.net/job/HMCTS/job/ccd-case-management-web/view/change-requests/job/PR-28/

Reply:
added test case